### PR TITLE
Add string methods: `isupper()` and `islower()`

### LINKED ISSUE #387: [Entry] Python: String Methods isupper() and islower()
## Description

I found that the existing docs on Python strings has entries for `lower()` and `upper()` methods. 
It would be good to add the related methods `islower()` and `isupper()` for checking if all characters in the string are in lower case or upper case, respectively.

## Type of Entry Request

- One or more new entries/terms requested.

## Checklist

- [x] The issue is appropriately titled.
- [x] Description is clear and includes supporting text, files, etc.
- [x] A similar issue hasn't already been created.
- [x] No more than 3 entries requested.

## For Maintainers

- [x] Labels are added
- [x] Assignee is added

### PR DESCRIPTION
**Pull request by balapriyac**
### Description

Add Python string methods `islower()` and `isupper()`. Closes #387 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by yangc95** [comment]
Hi @balapriyac! 👋 Review 1️⃣ will be done shortly

File: content/python/concepts/strings/terms/islower/islower.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Tags:
+  - 'Strings'
+  - 'Methods'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/analyze-data-with-python'
+---
+
+The `.islower()` string method takes in a string and returns `True` if all the letters in the string are in lowercase, else returns `False`. This method ignores spaces, newlines, numeric and special characters in the string.
```
Comment (yangc95) [inline content/python/concepts/strings/terms/islower/islower.md:16]:
```suggestion
The `.islower()` string method takes in a string and returns `True` if all the letters in the string are in lowercase, else returns `False`. This method ignores spaces, newlines, numeric, and special characters in the string.
```

File: content/python/concepts/strings/terms/isupper/isupper.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Use `.isupper()` to check if a string contains only uppercase letters:
+
+```py
+my_string = "Python3"
+
+print(my_string.isupper())
+# Output: False
+```
+
```
Comment (yangc95) [inline content/python/concepts/strings/terms/isupper/isupper.md:34]:
```suggestion
```

File: content/python/concepts/strings/terms/isupper/isupper.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Tags:
+  - 'Strings'
+  - 'Methods'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/analyze-data-with-python'
+---
+
+The `.isupper()` string method takes in a string and returns `True` if all the letters in the string are in uppercase, else returns `False`. This method ignores spaces, newlines, numeric and special characters in the string.
```
Comment (yangc95) [inline content/python/concepts/strings/terms/isupper/isupper.md:16]:
```suggestion
The `.isupper()` string method takes in a string and returns `True` if all the letters in the string are in uppercase, else returns `False`. This method ignores spaces, newlines, numeric, and special characters in the string.
```

**Review by yangc95 — APPROVED** [review]
❗ I see no real faults with either of these entries. @balapriyac I just suggested some missing commas and removed a space.

Going to update those and pass this along.

**Comment by SSwiniarski** [comment]
@balapriyac, I'll be doing review 2.

**Review by SSwiniarski — APPROVED** [review]
It all looks good to me.

**Comment by Dusch4593** [comment]
@balapriyac @SSwiniarski @yangc95 about to merge but made changes in 3e15c203a9726dfb3beecfb58b167bb78abaec82 and 78ea81875338584f653e28a0e3c0f44e89524aaf where I added some text in the `Codebyte Example` sections.

**Comment by yangc95** [comment]
@balapriyac We strongly encourage you to add a profile picture to your Codecademy profile so the entry has a visible contributor (although, if you can jog my memory, did you wish to stay anonymous?) ☺️ 

Screen Shot 2021-12-27 at 4 14 50 PM

Your entries are live here for reference:

`islower()`:
https://www.codecademy.com/resources/docs/python/strings/islower
`isupper()`:
https://www.codecademy.com/resources/docs/python/strings/isupper

**Comment by balapriyac** [comment]
Hi @yangc95, I had updated my Codecademy profile earlier, and have just added my GitHub username as well.
I'm not sure why it's showing `anonymous`. I guess it will take a while to show up on the docs page? 😊

**Comment by yangc95** [comment]
> Hi @yangc95, I had updated my Codecademy profile earlier, and have just added my GitHub username as well. I'm not sure why it's showing `anonymous`. I guess it will take a while to show up on the docs page? 😊

I hope that's the case! Though I'm wondering why I don't see your handle on the list of community contributors either. How does your privacy settings look?

Screen Shot 2021-12-28 at 2 25 36 PM

**Comment by yangc95** [comment]
@balapriyac checking back in ⏯️ any development?

**Comment by balapriyac** [comment]
Hi Christine, I'm not sure why I'm still `anonymous`. I'm able to view that
my Codecademy profile is visible to everyone (in my settings).

Earlier this year, when my hacktoberfest PRs were merged, I was able to see
my GitHub username in the docs that went live. Now, even those have the
tag, `Anonymous contributor`.

I'm not sure how this works. Is it like—I've updated my GitHub profile link
in my Codecademy profile. So only my subsequent PRs will be attributed to
me?

I thought maybe we should try that the next time and see if it's getting
updated. 😄

Happy new year to you. 🎉💫

**Comment by Dusch4593** [comment]
@balapriyac There may be some internal logic issue regarding how contributors are displayed on the entries. My colleagues and I just id'd another recently merged PR with the same issue as you. I just passed all this info along to my contacts in the Engineering team and we'll hopefully get some more insight into what's going on. I don't expect any updates until next week after the holidays are done, though. But I really appreciate your bringing this to our attention as we want you to get the credit you deserve!

**Comment by balapriyac** [comment]
Thank you very much, Brandon. Yes, I hadn't quite noticed this. It was
Christine who had identified this, and asked me if I wished to remain
anonymous as the contributor name wasn't showing up—that's how I got to
know of this. 🙂

Thanks again. We can revisit this early next year. No problem at all.

**Comment by Dusch4593** [comment]
Happy New Year @balapriyac! We have an update!

In addition to setting your Codecademy profile visibility to "Everyone", you have to enable SAML single sign-on ([SSO](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/about-authentication-with-saml-single-sign-on)) between your Codecademy and GitHub profiles _through_ the same primary email address. This can be done by following some steps [here in the CONTRIBUTING.md guide](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#codecademy-username).

Hope this helps but definitely let us know if you have any questions! 😃

**Comment by balapriyac** [comment]
Thank you very much @Dusch4593. I'll do that soon. 🙂

**Comment by balapriyac** [comment]
image
Thanks for the help @yangc95 and @Dusch4593, I'm now able to see my name beside my contributions.😄

**Pull request merged**